ft_atoi.c

```diff
@@ -10,20 +10,24 @@
 /*                                                                            */
 /* ************************************************************************** */
 
-int	ft_atoi(const char *str)
+int ft_atoi(const char *str)
 {
 	int sign;
 	int res;
 
 	res = 0;
 	sign = 1;
-	while (*str == ' ' || *str == '\f' || *str == '\r' || *str == '\t'
-			|| *str == '\v' || *str == '\n')
+	while (*str == ' ' ||
+				 *str == '\f' ||
+				 *str == '\r' ||
+				 *str == '\t' ||
+				 *str == '\v' ||
+				 *str == '\n' ||
+				 *str == '+')
 		str++;
-	if (*str == '-' || *str == '+')
```

Comment:
Well, if it had an && operator I would say it's !necessary but it's an || operator which means there's a test case where it failed so I added the statement x)

Reply:
Lol the only case would be if u add a "+" sign to the param and it just ignores it after all like atoi("+69") would output 69

Reply:
try ```printf("%d", atoi("+2147483648")); ``` and compare it with ```printf("%d", ft_atoi("+2147483648")); ``` :D I don't add a single line to my code without purpose :D

Reply:
Well I just thought bout another case lol atoi("+-69") according to my changes I think it will output -69 while the other would return 0.

Reply:
both will return 0 x) with or without change, my memory is not perfect but If I remember well, I added the statement because of the test that Ipreviously mentioned. 
The max range of 4-byte integer is 2147483647 and in the test ```printf("%d", ft_atoi("+2147483648"));``` I'm kinda trying to fool ft_atoi and without the statement, it will return 0.
If you want to know the reason why you can print ```res``` and you'll notice the change that happens during the conversion.

Reply:
well I don't think so 

![image](https://user-images.githubusercontent.com/19322270/100470429-86594a80-30d8-11eb-86c8-39c2fc47902b.png)

I just tried it, it's the same output

Reply:
well as I said, "without the statement, it will return 0." I meant with the changes that you made ft_atoi will return 0 in such case x)

Reply:
lol nah it doesn't, that capture is with my changes __included__ XDs

Reply:
well you probably made some mistakes there or you're drunk xD here's the code with the changes you made:
```C
#include <stdio.h>
#include <stdlib.h>

int	ft_atoi(const char *str)
{
	int sign;
	int res;

	res = 0;
	sign = 1;
	while (*str == ' ' || *str == '\f' || *str == '\r' || *str == '\t'
			|| *str == '\v' || *str == '\n')
		str++;
	if (*str == '-')
	{
	    sign = -sign;
		str++;
	}
	while (*str != '\0' && (*str >= '0' && *str <= '9'))
	{
		if (res < 0 && sign < 0)
			return (0);
		if (res < 0 && sign > 0)
			return (-1);
		res = (res * 10) + (*str - '0');
		str++;
	}
	return (res * sign);
}

int main()
{
    printf("%d\n", ft_atoi("+2147483648"));
    printf("%d", atoi("+2147483648"));

    return 0;
}
```
the output is:
![image](https://user-images.githubusercontent.com/26191219/100477847-db518c80-30e9-11eb-9582-06cb18131fbe.png)